### ISSUE TITLE
Error when server closing

### ISSUE THREAD

**Issue by Yurinann** [PaperSpigot/Paper#7495]
### Stack trace

```
[20:56:31] [Paper Async Task Handler Thread - 176/ERROR]: Caught previously unhandled exception :
[20:56:31] [Paper Async Task Handler Thread - 176/ERROR]: Paper Async Task Handler Thread - 176
java.util.concurrent.RejectedExecutionException: null
	at java.util.concurrent.ForkJoinPool.externalPush(ForkJoinPool.java:2179) ~[?:?]
	at java.util.concurrent.ForkJoinPool.externalSubmit(ForkJoinPool.java:2196) ~[?:?]
	at java.util.concurrent.ForkJoinPool.execute(ForkJoinPool.java:2664) ~[?:?]
	at java.util.concurrent.CompletableFuture.asyncSupplyStage(CompletableFuture.java:1782) ~[?:?]
	at java.util.concurrent.CompletableFuture.supplyAsync(CompletableFuture.java:2005) ~[?:?]
	at me.lucko.luckperms.common.storage.Storage.future(Storage.java:86) ~[?:?]
	at me.lucko.luckperms.common.storage.Storage.loadUser(Storage.java:148) ~[?:?]
	at me.lucko.luckperms.bukkit.vault.LuckPermsVaultPermission.lookupUser(LuckPermsVaultPermission.java:159) ~[?:?]
	at me.lucko.luckperms.bukkit.vault.LuckPermsVaultPermission.userHasPermission(LuckPermsVaultPermission.java:174) ~[?:?]
	at me.lucko.luckperms.bukkit.vault.AbstractVaultPermission.playerHas(AbstractVaultPermission.java:151) ~[?:?]
	at com.sk89q.wepif.VaultResolver.hasPermission(VaultResolver.java:100) ~[worldedit-bukkit-7.2.9.jar:?]
	at com.sk89q.wepif.PermissionsResolverManager.hasPermission(PermissionsResolverManager.java:170) ~[worldedit-bukkit-7.2.9.jar:?]
	at com.sk89q.worldedit.bukkit.BukkitPlayer.hasPermission(BukkitPlayer.java:174) ~[worldedit-bukkit-7.2.9.jar:?]
	at java.util.stream.MatchOps$1MatchSink.accept(MatchOps.java:90) ~[?:?]
	at java.util.Spliterators$IteratorSpliterator.tryAdvance(Spliterators.java:1856) ~[?:?]
	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:129) ~[?:?]
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:527) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:513) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:230) ~[?:?]
	at java.util.stream.MatchOps$MatchOp.evaluateSequential(MatchOps.java:196) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.anyMatch(ReferencePipeline.java:632) ~[?:?]
	at com.sk89q.worldedit.command.util.PermissionCondition.lambda$satisfied$0(PermissionCondition.java:47) ~[worldedit-bukkit-7.2.9.jar:?]
	at java.util.Optional.map(Optional.java:260) ~[?:?]
	at com.sk89q.worldedit.command.util.PermissionCondition.satisfied(PermissionCondition.java:47) ~[worldedit-bukkit-7.2.9.jar:?]
	at com.sk89q.worldedit.bukkit.BukkitCommandInspector.testPermission(BukkitCommandInspector.java:81) ~[worldedit-bukkit-7.2.9.jar:?]
	at com.sk89q.bukkit.util.DynamicPluginCommand.testPermissionSilent(DynamicPluginCommand.java:100) ~[worldedit-bukkit-7.2.9.jar:?]
	at org.bukkit.craftbukkit.v1_18_R1.command.BukkitCommandWrapper.test(BukkitCommandWrapper.java:48) ~[purpur-1.18.1.jar:git-Purpur-1562]
	at org.bukkit.craftbukkit.v1_18_R1.command.BukkitCommandWrapper.test(BukkitCommandWrapper.java:20) ~[purpur-1.18.1.jar:git-Purpur-1562]
	at com.mojang.brigadier.tree.CommandNode.canUse(CommandNode.java:79) ~[purpur-1.18.1.jar:git-Purpur-1562]
	at net.minecraft.commands.Commands.fillUsableCommands(Commands.java:423) ~[?:?]
	at net.minecraft.commands.Commands.sendAsync(Commands.java:384) ~[?:?]
	at net.minecraft.commands.Commands.sendCommands(Commands.java:368) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635) ~[?:?]
	at java.lang.Thread.run(Thread.java:833) ~[?:?]
```

### Plugin and Datapack List

> plugins
[21:04:26 INFO]: Plugins (51): AdvancedCleaner, AntiCombatLog, AureliumSkills, AuthMe, BentoBox, BetterEnchantBook*, Citizens, CMI, CMILib, CrazyAuctionsPlus, CustomCobbleGen, DeathLostMoney*, DecentHolograms, EastLandLog4jFixer*, ExcellentCrates, FaithlPack, InteractionVisualizer, JonAPI*, LightAPI, LiteBans, LiteSignIn, LuckPerms, MineResetLite, MiniMOTD, Multiverse-Core, NBTAPI, NexEngine, PlaceholderAPI, PlayerPoints, PlugManX (PlugMan), PointLogger, PPRMB, ProtocolLib, PVPRewards, questscrolls*, QuickShop, RainbowGzExpansion, SkinsRestorer, SpawnerMeta, SurpriseFood, TAB, TrChat, TrMenu, UserPrefix, UTitleAuth, Vault, WorldBorderAPI, WorldEdit, WorldGuard, XConomy, YukiNoaAPI
> datapack list
[21:04:27 INFO]: There are 2 data packs enabled: [vanilla (built-in)], [file/bukkit (world)]
[21:04:27 INFO]: There are no more data packs available

### Actions to reproduce (if known)

use /stop to close server

### Paper version

Checking version, please wait...
Current: git-Purpur-1562 (MC: 1.18.1)*

(Paper also)

**Comment by kennytv** [comment]
This is not a Paper but a plugin issue (nor are you actually running Paper)

**Comment by AlistairWstbrk** [comment]
So what plugin was causing the issue I have seen the same error.

**Comment by willkroboth** [comment]
That stack trace is kinda a mess of potential error sources. We've got `craftbukkit.v1_18_R1` via `purpur-1.18.1.jar:git-Purpur-1562`, `com.sk89q.worldedit`, `me.lucko.luckperms`, and the Paper Async Task Handler all leading up to a kinda mysterious `RejectedExecutionException`. If I had to guess I would say that WorldEdit didn't shut down its tasks correctly and it should be reported to them, but I'd need at least a server log to be certain myself.
